scRNA-seq-advanced/setup/intro-diff-expression-setup.Rmd

```diff
@@ -0,0 +1,624 @@
+
+Below are a few reasons why pseudo-bulking is helpful: 
+
+- Produces larger counts which are more amenable to standard normalization and differential expression methods used by bulk RNA-sequencing. 
+- Collapses gene expression counts by sample, reflecting that each sample is a replicate rather than each cell. 
+- Masks variance within a sample and only accounts for if a gene is uniformly high or low across samples. 
+
+Let's start with the simple comparison of looking at one cell type and comparing across ERMS and ARMS subtypes. 
+When we do this, we will first want to pseudo-bulk our dataset to group our cells by cell type and by sample.
+Then we can subset the pseudo-bulked dataset to just contain our cell type of interest before proceeding. 
+
+```{r}
+# first let's subset our coldata to only have the columns we care about in pseudobulking 
+pseudobulked_coldata <- colData(rms_sce)[, c("celltype_broad", "batch")]
+
+# this creates a new SCE object that contains the pseudobulked counts across the provided groups 
+pseudobulked_sce <- scater::aggregateAcrossCells(rms_sce, 
+                                                  id = pseudobulked_coldata)
+
+pseudobulked_sce
+```
+```{r}
+# only 37 columns in the counts assay 
+counts(pseudobulked_sce)[1:10, ]
+```
+
+Let's take a look at what the `colData` now looks like in the pseudo-bulked SCE object. 
+
+```{r}
+# note the new addition of the column with number of cells per group 
+colData(pseudobulked_sce)
+
+```
+
+Now we filter to just the sub type of cells that we are interested in. 
+
+```{r}
+tumor_myoblast_sce <- pseudobulked_sce[, pseudobulked_sce$celltype_broad == "Tumor_Myoblast"]
+table(tumor_myoblast_sce$celltype_broad)
+```
+
+We should now see that there are only 6 samples left that all have the cell type `Tumor_Myoblast`. 
+
+Before we can perform DGE analysis, we want to do some quality control like filtering out any groups in the pseudo-bulked dataset that may have a low number of cells. 
+
+```{r}
+# filter by number of cells 
+low_cell_groups <- tumor_myoblast_sce$ncells < 10
+tumor_myoblast_sce <- tumor_myoblast_sce[, !low_cell_groups]
+```
+
+We also want to filter out any genes that may be lowly expressed in our pseudo-bulked dataset.
+
+```{r}
+# filter out genes that might be lowly expressed using filterbyExpr in `edgeR`
+# provide sample groups from SCE to determine minimum sample size that genes should be expressed in 
+# we are grouping samples as ERMS vs. ARMS so use subdiagnosis
+genes_filter <- edgeR::filterByExpr(tumor_myoblast_sce,
+                                     group = tumor_myoblast_sce$subdiagnosis)
+tumor_myoblast_sce <- tumor_myoblast_sce[genes_filter,]
+
+# looks like this doesn't actually filter anything though, but we probably want to keep this step in? it seems important 
```

Comment:
This may be a good teachable to talk about the function defaults and whether they are reasonable for our libraries. Relevant defaults are - 
 - `min.count = 10`: Minimum count required for at least some samples.
 - `min.total.count = 15`: Minimum total count required.
 - `large.n = 10`: Number of samples per group that is considered to be “large”. 
 - `min.prop = 0.7`: Minimum proportion of samples in the smallest group that express the gene.